design/pool-configuration.md

```diff
@@ -0,0 +1,386 @@
+# Address Pool configuration options
+
+## Summary
+
+Following @johananl's initiative in [this issue](https://github.com/metallb/metallb/issues/630)
+this enhancement tries to collect and classify various configuration degrees requested during time.
+
+## Motivation
+
+The way advanced users consume MetalLB is evolving and exposed a need for more complex configuration.
+For example, the need for isolation and multi-tenancy induced requests for being able to advertise some specific address pools from a subset of the nodes, or
+to allocate them only to a specific namespace, to mention some.
+
+### Goals
+
+* Collect and classify all the requests coming from different issues, in order to have a better understanding
+  and an organic implementation, avoiding to focus on and to implement only a specific feature, risking to cause conflicts afterwards.
+* Provide a list of the possible configuration enhancements, and to highlight dependencies and conflicts between each other (if any).
+* Highlight the cases where the solution must be BGP or L2 specific
+
+### Non-Goals
+
+* Covering only a specific use case.
+* Providing a L2 only or BGP only solution where the problem might be expressed in a more generic way.
+
+## Proposal
+
+For each ip of an address pool, we need to address three main points:
```

Comment:
This phrasing is ambiguous IMO.

Let IP A and IP B be IPs in address pool X. Which of the following is true?

1. IP A can be allocated/advertised separately from IP B. For example, the user can decide that IP A be advertised only from node A whereas IP B be advertised from all available nodes.
2. IP A and IP B are always treated the same way. IOW, the policy options which control IP allocation/advertisement apply to **all IPs** in a pool.

Personally, I think option 2 makes the most sense. If you want to treat an individual IP in a special way, you can always define a pool with just that IP.

Reply:
Yep, I agree and that's what I tried to express.
I'll rephrase to make it clear that any behavior we introduce will be consistent for all the IPs of a given pool.